### DISCUSSION TITLE
Change the structure of if-command in level 4

### DISCUSSION THREAD

**Discussion by MarleenGilsing** [hedyorg/hedy#901] [Language Ideas]
@Felienne and I were talking about the structure of the if-command.

We started out with:
if name is Hedy print 'fun' else ' print 'not fun'

We've already changed it to: 
if name is Hedy print 'fun' 
else print 'not fun'

But we'd like to change it to:

if name is Hedy
print 'fun' 
else 
print 'not fun'

This has the advantage that it looks more like Python structure and it makes the lines of code shorter and easier to oversee. 
It does make it harder to see that these lines are a block and belong together. That's why we thought it would be a good idea to teach the students to use enters before and after this block to make it a block visually. (But Hedy will also work if you don't use the enters). So it will look like this: 

print 'Hello Im Hedy'
name is ask 'Whats your name?'

if name is hedy
print 'fun'
else
print 'not fun'

print 'Nice to meet you!'
print 'Bye now.'

@OnnoEbbens What do you think of this change?

**Comment by OnnoEbbens** [comment]
For me it is a lot more clear with this change. However I am not so sure if it better for the students. 

Some thought I have:
- The current syntax for the if-command in level 4 reads more like a real sentence. With the extra enters it looks more like code. I think the principle of Hedy is to start with code that looks like real language and change gradually to more code-like syntax.
- Now you can only use an if-statement with one line in level 4. The students understand that and think it is annoying. This gives them a big incentive to learn level 7 where we introduce indentation, now they can finally make multi-line if-statements and use the `ask` command within the if.
- The current syntax of the if-statement in level 4 is confusing sometimes. Especially if you create a variable inside an if-statement, e.g. `if name is Bond name2 is James`. This would be a good reason to implement the new syntax. On the other hand we can also limit the if-statement in level 4 in such a way that you can only use print, forward and turn within the if/else. Then in level 7 we can introduce the other commands within the if-statement. This might also make it easier to allow a check in the if-statement with two words (see #856).

So not a big yes or no for me.

**Reply by MarleenGilsing** [reply]
You make a good point that the current syntax reads more like a sentence and would therefore be more natural to the students. On the other hand, the sentence is too long and doesn't use a comma like you would in a long sentence of normal language. So it's still hard to read eventough it looks like a normal sentence. 

We did intend to keep the if-statement with only one line in level 4, despite the new lay-out. So that would still encourage the students to get to level 7.
